lxc/storage_volume.go

```diff
@@ -189,26 +189,49 @@ func (c *cmdStorageVolumeAttach) Run(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf(i18n.G("Missing pool name"))
 	}
 
+	volName, volType := c.storageVolume.parseVolume("custom", args[1])
+	if volType != "custom" {
+		return fmt.Errorf(i18n.G("Only \"custom\" volumes can be attached to instances"))
+	}
+
 	// Attach the volume
 	devPath := ""
 	devName := ""
 	if len(args) == 3 {
 		devName = args[1]
 	} else if len(args) == 4 {
-		// Only the path has been given to us.
-		devPath = args[3]
-		devName = args[1]
+		client := resource.server
+
+		// Use the provided target.
+		if c.storage.flagTarget != "" && client.IsClustered() {
+			client = client.UseTarget(c.storage.flagTarget)
+		}
+
+		vol, _, err := client.GetStoragePoolVolume(resource.name, volType, volName)
+		if err != nil {
+			return err
+		}
+
+		switch vol.ContentType {
+		case "block", "iso":
+			devName = args[3]
+		case "filesystem":
+			if !strings.HasPrefix(args[3], "/") {
```

Comment:
Why is this needed? If it is needed it should be in a separate commit as not related to the commit message.

Reply:
This is probably my [suggestion](https://github.com/canonical/lxd/pull/13258#issuecomment-2040276569), I think it makes sense to append `/` on the start of the path to better convey where the device is being mounted (showing `/something` instead of `something`). So it is indeed unrelated to the commit message.